pdaltools/pcd_info.py

```diff
@@ -0,0 +1,46 @@
+"""Tools to get information from a point cloud (points as a numpy array)"""
+
+from typing import Tuple
+
+import numpy as np
+
+
+def get_pointcloud_origin_from_tile_width(points: np.array, tile_width: int = 1000, buffer_size: float = 0) -> Tuple:
+    """Get point cloud theoretical origin (xmin, ymax) for a data that originates from a square tesselation/tiling
+    using the tesselation tile width only.
+
+    Edge values are supposed to be included in the tile
+
+
+    Args:
+        points (np.array): numpy array with the tile points
+        tile_width (int, optional): Edge size of the square used for tiling. Defaults to 1000.
+        buffer_size (float, optional): Optional buffer around the tile. Defaults to 0.
+
+    Raises:
+        ValueError: Raise an error when the bounding box of the tile is not included in a tile
+
+    Returns:
+        Tuple: (origin_x, origin_y) origin coordinates
+    """
+    # Extract coordinates xmin, xmax, ymin and ymax of the original tile without buffer
+    x_min, y_min = np.min(points[:, :2], axis=0) + buffer_size
+    x_max, y_max = np.max(points[:, :2], axis=0) - buffer_size
+
+    # Calculate the difference Xmin and Xmax, then Ymin and Ymax
+    diff_tile_x = np.floor(x_max / tile_width) - np.floor(x_min / tile_width)
+    is_x_max_on_edge = x_max % tile_width == 0
```

Comment:
Si tu ne veux pas t'embêter avec la variable is_x_max_on_edge, qui impose un if un peu velu plus bas, il est possible de faire :
`x_max = x_max - tile_width if x_max % tile_width ==0 else x_max`
`diff_tile_x = np.floor(x_max / tile_width) - np.floor(x_min / tile_width)`

et plus bas il suffit de tester
`if not diff_tile_x and not diff_tile_y:`

Reply:
j'ai une proposition intermédiaire pour tenter de garder le sens des différentes variables, ça te parait mieux ?